packages/roosterjs-editor-core/lib/coreApi/getContent.ts

```diff
@@ -48,13 +49,14 @@ export const getContent: GetContent = (core: EditorCore, mode: GetContentMode):
         const range = path && createRange(clonedRoot, path.start, path.end);
 
         if (core.lifecycle.isDarkMode) {
-            core.api.transformColor(
-                core,
-                clonedRoot,
-                false /*includeSelf*/,
-                null /*callback*/,
-                ColorTransformDirection.DarkToLight
-            );
+            const event: ColorChangedEvent = {
```

Comment:
Now I think I understand the problem.

So there are two different scenarios:
1. A format API did some color change, and need the plugin do color transform if editor is in dark mode. This happens in:
   - Apply table format
   - Paste
   - insertNode
   - replaceNode
   - setTextColor and setBackgroundColor
   - setContent
2. Need to do a color transformation to a given direction so that we get the html content:
   - getContent

So we need the plugin for 1, because it can be triggered from multiple sources and we don't want all of them care about current color mode and the detail of dark mode.

But for 2, we just need a util function since it has very limited source and we clearly know the color mode.

So the general idea will be:
1. Keep the new plugin
2. Simplify the event to only take necessary parameters. It doesn't need to know the direction since this plugin will only do light to dark transformation.
3. Move transformColor function out to be a single function (possibly in dom package), then both the plugin and getContent() can call it to do color transformation.
4. Trigger ColorChanged event for those APIs to let plugin handle color transformation in dark mode.